### PR TITLE
 [Doctest] Add `configuration_clip.py`

### PR DESCRIPTION
**Pull request by daspartho**
Add `configuration_clip.py` to `utils/documentation_tests.txt` for doctest.

Based on issue #19487

Noticed the model initialization and config initialization lines were switched so made some additional changes to correct it.

@ydshieh could you please check if it's okay?
Thank you =)

### CODE REVIEW

**Comment by daspartho** [comment]
@ydshieh Sure! I'd like to take on the task =)

**Comment by daspartho** [comment]
@ydshieh made some changes; could you please check if it looks good?

**Comment by daspartho** [comment]
@ydshieh added an example using the `from_text_vision_configs` method; could you please review the changes to see if they're okay? 
Thanks :)

File: src/transformers/models/clip/configuration_clip.py [hunk 1]
```diff
@@ -270,7 +270,27 @@ class CLIPConfig(PretrainedConfig):
...
+
+    >>> # Initializing a CLIPConfig with openai/clip-vit-base-patch32 style configuration
+    >>> configuration = CLIPConfig()
+
+    >>> # Initializing a CLIPModel (with random weights) from the openai/clip-vit-base-patch32 style configuration
+    >>> model = CLIPModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+
```
Comment (ydshieh) [inline src/transformers/models/clip/configuration_clip.py:287]: Let's add a comment to explain the objective here, like

`# We can also initialize a CLIPConfig from a CLIPTextConfig and a CLIPVisionConfig`

**Comment by ydshieh** [comment]
Just a final comment and we are ready to merge!

**Comment by daspartho** [comment]
@ydshieh made the suggested changes; good to go =)

**Comment by ydshieh** [comment]
Thank you for the PR and your patience :-) @daspartho

**Review by ydshieh — APPROVED** [review]

**Pull request merged**